platforms/ide/problems-api/src/main/java/org/gradle/api/problems/AdditionalData.java

```diff
@@ -14,10 +14,14 @@
  * limitations under the License.
  */
 
-package org.gradle.api.problems.internal;
+package org.gradle.api.problems;
 
 import org.gradle.api.Action;
 import org.gradle.api.Incubating;
+import org.gradle.api.problems.internal.InternalProblemSpec;
+import org.gradle.api.problems.internal.Problem;
+
+import java.io.Serializable;
 
 /**
  * Marker interface for additional data that can be attached to a {@link Problem}.
```

Comment:
We should be very explicit on the contract here. 
-Why we require serialisable here
   - will be sent from the daemon to the consumers 
- What happens with the report if the serialization fails?

Alternatively, please create a follow-up issue for documentation.